### PR TITLE
rename *_data.c to *_data.h

### PR DESCRIPTION
**Pull request by bluesquall**
If you're not going to `#include` a header with the macro, you need to `#define` it in the source file.

I noticed this when `make check` failed in [my `autotools` branch](https://github.com/bluesquall/GSW-C/tree/autotools) after merging recent changes from upstream. I suspect you don't experience it with your manual Makefile because of the specific way you call `gcc`. But if you want to be able to compile `gsw_saar_data.c -> gsw_saar_data.o` without implicitly depending on additional files, you need to define the macro within.

You would also use a `#ifndef ... #define ... #endif` block, if you'd prefer.

### CODE REVIEW

**Comment by efiring** [comment]
It doesn't work with the Windows compiler; see how the definition is handled in gsw_internal_const.h.  But beyond that, we are simply including gsw_saar_data in gsw_saar, which works fine.  We don't need or use or make gsw_saar_data.o.

**Comment by bluesquall** [comment]
Interesting … I will look again later and see if I can come up with a solution that works with autotools (or Cmake, or another build system) and with your windows test.

Using a build system is critical for my sanity when deploying as a dependency on various systems. I went with autotools first because one of the systems I have needs to be cross-compiled and I had other examples to use for reference, but I’m open to alternatives in general.

**Comment by efiring** [comment]
Would the problem be solved by renaming gsw_saar_data.c to gsw_saar_data.h, and continuing to #include it as we do now?  Is the problem that autotools assumes any .c file will be compiled to .o rather than inserted via #include?

**Comment by bluesquall** [comment]
That’s a good, simple approach. I will try it later today when I get back into code.

**Comment by bluesquall** [comment]
Tested with my autotools branch and it seems to have worked.

It looks like the windows test only runs on a PR with approval from a maintainer.

File: gsw_saar_data.h [hunk 1]
```diff
@@ -2,6 +2,7 @@
 **  $Id$
 **  Extracted from ../../gsw_matlab_v3_06_11/library/gsw_data_v3_0.mat
 */
+#define UNUSED __attribute__ ((unused))
```
Comment (efiring) [inline gsw_saar_data.h:5]: This line should be removed now.

**Review by efiring — CHANGES_REQUESTED** [review]
I think this will be fine once you remove the #define statement.

**Comment by bluesquall** [comment]
Huh. I thought I got that before I pushed the rebased changes. I'll check again.

**Review by efiring — APPROVED** [review]

**Comment by efiring** [comment]
Thank you!

**Pull request merged**